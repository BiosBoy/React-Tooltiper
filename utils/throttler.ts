interface IThrottler {
  event: string
  callback: Function
  delay?: number
  target?: Window
}

const throttler = ({ event = 'resize', callback = () => {}, delay = 50, target = window }: IThrottler) => {
  let timerID = null
  const eventSubscribeTarget = target || window

  const resizeThrottler = () => {
    if (timerID) return

    timerID = setTimeout(() => {
      callback()
      timerID = null
    },                   delay)
  }

  eventSubscribeTarget.addEventListener(event, resizeThrottler, false)
}

export default throttler
