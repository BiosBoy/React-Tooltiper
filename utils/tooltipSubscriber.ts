import { ITooltips } from '../types'

class ActionObserver {
  actions?: Function[]

  constructor() {
    this.actions = []
  }

  unitAlreadyAdded = unit => this.actions.some(func => func.name === unit.name)

  addAction = (unit: Function) => {
    if (this.unitAlreadyAdded(unit)) return

    this.actions.push(unit)
  }

  removeAction = (f: Function) => {
    this.actions = this.actions.filter(unit => unit !== f)
  }

  notifyActions = (payload: ITooltips[]) => {
    this.actions.forEach(unit => unit(payload))
  }
}

class TooltipObserver extends ActionObserver {
  targets?: ITooltips[]

  constructor() {
    super()

    this.targets = []
  }

  getTargets = () => this.targets

  subscribe = (target: ITooltips) => {
    const targetAlreadyIn = this.targets.some(item => item.ID === target.ID)

    if (targetAlreadyIn) return

    this.targets.push(target)
    this.notifyActions(this.targets)
  }

  update = (target: ITooltips) => {
    this.targets.forEach((item, index) => {
      const targetMatch = item.ID === target.ID
      const childsNotEqual = item.child !== target.child

      if (targetMatch && childsNotEqual) {
        this.targets[index] = {
          ...item,
          child: target.child
        }

        this.notifyActions(this.targets)
      }
    })
  }

  unsubscribe = (target: ITooltips) => {
    this.targets = this.targets.filter(item => item.ID !== target.ID)
    this.notifyActions(this.targets)
  }

  destroyAll = () => {
    this.targets.splice(0, this.targets.length)
    this.actions.splice(0, this.actions.length)
  }
}

const tooltipSubscriber = new TooltipObserver()

export default tooltipSubscriber
