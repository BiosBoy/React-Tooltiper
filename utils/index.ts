import tooltipSubscriber from './tooltipSubscriber'
import throttler from './throttler'

export { tooltipSubscriber, throttler }
