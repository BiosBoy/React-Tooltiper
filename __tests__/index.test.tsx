import React from 'react'
import { mount } from 'enzyme'

import initialState, { withoutArrow, closeButton, positionCoords, customCoords, manualCoodsFix } from './mocks'
import Tooltip from '..'
import { tooltipSubscriber } from '../utils'

describe('<Tooltip test />', () => {
  beforeEach(() => {
    Element.prototype.getBoundingClientRect = jest.fn(() => {
      return {
        width: 120,
        height: 120,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0
      }
    })
  })

  it('should render basic Tooltip with Title and Arrow included', () => {
    const Wrapper = mount(<div id='tooltip-test' />)
    const Component = mount(<Tooltip {...initialState} />)

    Component.setState({
      showTooltip: true,
      tooltipNode: Component.find('ToolTip'),
      parentNode: Wrapper.find('#tooltip-test'),
      tooltip: { child: 'TEST TOOLTIP', ID: 'tooltip-test' }
    })
    Component.update()

    tooltipSubscriber.subscribe({ child: 'TEST TOOLTIP', ID: 'tooltip-test' })

    Wrapper.find('#tooltip-test').simulate('mouseOver')
    Wrapper.find('#tooltip-test').simulate('mouseover')

    expect(Component.find('ToolTip').length).toBe(1)
    expect(Component.find('.tooltipContainer').length).toBe(1)
    expect(Component.find('.title').length).toBe(1)
    expect(Component.find('.title').text()).toBe('TEST TOOLTIP')
    expect(Component.find('.tooltipArrow').length).toBe(1)

    expect(Component).toMatchSnapshot()
  })
  it('should not render Tooltip in case of showTooltip = false flag ', () => {
    const Component = mount(<Tooltip {...initialState} />)

    Component.setState({
      showTooltip: false,
      tooltipNode: null,
      parentNode: null,
      tooltip: { child: '', ID: '' }
    })
    Component.update()

    expect(Component.find('ToolTip').length).toBe(1)
    expect(Component.find('.tooltipContainer').length).toBe(0)
    expect(Component.find('.title').length).toBe(0)
    expect(Component.find('.tooltipArrow').length).toBe(0)

    expect(Component).toMatchSnapshot()
  })
  it('should render Tooltip without Arrow', () => {
    const Wrapper = mount(<div id='tooltip-test' />)
    const Component = mount(<Tooltip {...withoutArrow} />)

    Component.setState({
      showTooltip: true,
      tooltipNode: Component.find('ToolTip'),
      parentNode: Wrapper.find('#tooltip-test'),
      tooltip: { child: 'TEST TOOLTIP', ID: 'tooltip-test' }
    })

    Component.update()

    expect(Component.find('ToolTip').length).toBe(1)
    expect(Component.find('.tooltipContainer').length).toBe(1)
    expect(Component.find('.title').length).toBe(1)
    expect(Component.find('.tooltipArrow').length).toBe(0)

    expect(Component).toMatchSnapshot()
  })
  it('should render Tooltip with Close Button', () => {
    const Wrapper = mount(<div id='tooltip-test' />)
    const Component = mount(<Tooltip {...closeButton} />)

    Component.setState({
      showTooltip: true,
      tooltipNode: Component.find('ToolTip'),
      parentNode: Wrapper.find('#tooltip-test'),
      tooltip: { child: 'TEST TOOLTIP', ID: 'tooltip-test' }
    })

    Component.update()

    expect(Component.find('ToolTip').length).toBe(1)
    expect(Component.find('.tooltipContainer').length).toBe(1)
    expect(Component.find('CloseButton').length).toBe(1)
    expect(Component.find('.closeButton').text()).toBe('XXX')
    expect(Component.find('.title').length).toBe(1)
    expect(Component.find('.tooltipArrow').length).toBe(1)

    expect(Component).toMatchSnapshot()
  })
  it('should render Tooltip with Close Button. By click on it should remove Tooltip at all', () => {
    const Wrapper = mount(<div id='tooltip-test' />)
    const Component = mount(<Tooltip {...closeButton} />)

    Component.setState({
      showTooltip: true,
      tooltipNode: Component.find('ToolTip'),
      parentNode: Wrapper.find('#tooltip-test'),
      tooltip: { child: 'TEST TOOLTIP', ID: 'tooltip-test' }
    })

    Component.update()

    expect(Component.find('ToolTip').length).toBe(1)
    expect(Component.find('.tooltipContainer').length).toBe(1)
    expect(Component.find('CloseButton').length).toBe(1)
    expect(Component.find('.closeButton').text()).toBe('XXX')
    expect(Component.find('.title').length).toBe(1)
    expect(Component.find('.tooltipArrow').length).toBe(1)

    expect(Component).toMatchSnapshot()

    Component.find('.closeButton').simulate('click')
    Component.setState({
      showTooltip: false,
      tooltipNode: null,
      parentNode: null,
      tooltip: { child: '', ID: '' }
    })
    Component.update()

    expect(Component.state('showTooltip')).toBe(false)
    expect(Component.find('ToolTip').length).toBe(1)
    expect(Component.find('CSSTransition').length).toBe(1)
    expect(Component.find('CSSTransition').prop('in')).toBe(false)

    expect(Component).toMatchSnapshot()
  })
  it('should render Tooltip by position props', () => {
    const Wrapper = mount(<div id='tooltip-test' />)
    const Component = mount(<Tooltip {...positionCoords} />)

    Component.setState({
      showTooltip: true,
      tooltipNode: Component.find('ToolTip'),
      parentNode: Wrapper.find('#tooltip-test'),
      tooltip: { child: 'TEST TOOLTIP', ID: 'tooltip-test' }
    })

    Component.update()

    expect(Component.find('ToolTip').length).toBe(1)
    expect(Component.find('.tooltipContainer').length).toBe(1)
    expect(Component.prop('position')).toEqual({ x: 'center', y: 'top' })

    expect(Component).toMatchSnapshot()
  })
  it('should render Tooltip by custom coords props', () => {
    const Wrapper = mount(<div id='tooltip-test' />)
    const Component = mount(<Tooltip {...customCoords} />)

    Component.setState({
      showTooltip: true,
      tooltipNode: Component.find('ToolTip'),
      parentNode: Wrapper.find('#tooltip-test'),
      tooltip: { child: 'TEST TOOLTIP', ID: 'tooltip-test' }
    })

    Component.update()

    expect(Component.find('ToolTip').length).toBe(1)
    expect(Component.find('.tooltipContainer').length).toBe(1)
    expect(Component.prop('customCoords')).toEqual({ top: '105px', left: '20px' })

    expect(Component).toMatchSnapshot()
  })
  it('should render Tooltip by manual coords coords fix props', () => {
    const Wrapper = mount(<div id='tooltip-test' />)
    const Component = mount(<Tooltip {...manualCoodsFix} />)

    Component.setState({
      showTooltip: true,
      tooltipNode: Component.find('ToolTip'),
      parentNode: Wrapper.find('#tooltip-test'),
      tooltip: { child: 'TEST TOOLTIP', ID: 'tooltip-test' }
    })

    Component.update()

    expect(Component.find('ToolTip').length).toBe(1)
    expect(Component.find('.tooltipContainer').length).toBe(1)
    expect(Component.prop('manualCoodsFix')).toEqual({ fixX: -2, fixY: -3 })

    expect(Component).toMatchSnapshot()
  })
})
