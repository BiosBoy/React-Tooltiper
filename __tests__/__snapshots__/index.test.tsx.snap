// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip test /> should not render Tooltip in case of showTooltip = false flag  1`] = `
<ToolTip
  actionOnEntered={[Function]}
  actionOnExit={[Function]}
  addArrow={true}
  animProps={
    Object {
      "className": "tooltip",
      "timeExit": 1000,
      "timeIn": 30,
    }
  }
  closeButton={
    Object {
      "actionOnClose": [Function],
      "active": false,
      "buttonChildren": "",
    }
  }
  customCoords={
    Object {
      "active": false,
      "left": "",
      "top": "",
    }
  }
  overrideStyles={
    Object {
      "arrowWrap": "",
      "button": "",
      "container": "",
      "title": "",
      "tooltipArrow": "",
      "wrap": "",
    }
  }
  position={
    Object {
      "x": "center",
      "y": "top",
    }
  }
>
  <CSSTransition
    classNames="tooltip"
    in={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onExited={[Function]}
    timeout={
      Object {
        "enter": 30,
        "exit": 1000,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 30,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    />
  </CSSTransition>
</ToolTip>
`;

exports[`<Tooltip test /> should render Tooltip by custom coords props 1`] = `
<ToolTip
  actionOnEntered={[Function]}
  actionOnExit={[Function]}
  addArrow={true}
  animProps={
    Object {
      "className": "tooltip",
      "timeExit": 1000,
      "timeIn": 30,
    }
  }
  closeButton={
    Object {
      "actionOnClose": [Function],
      "active": false,
      "buttonChildren": "",
    }
  }
  customCoords={
    Object {
      "left": "20px",
      "top": "105px",
    }
  }
  overrideStyles={
    Object {
      "arrowWrap": "",
      "button": "",
      "container": "",
      "title": "",
      "tooltipArrow": "",
      "wrap": "",
    }
  }
  position={
    Object {
      "x": "center",
      "y": "top",
    }
  }
>
  <CSSTransition
    classNames="tooltip"
    in={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onExited={[Function]}
    timeout={
      Object {
        "enter": 30,
        "exit": 1000,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 30,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <div
        className="tooltipWrap "
        id="#react-tooltip-tooltip-test"
        style={
          Object {
            "flexDirection": "column",
            "left": 0,
            "pointerEvents": "none",
            "top": -120,
            "transform": "translate(-50%, 0%)",
          }
        }
      >
        <div
          className=" tooltipContainer"
          key="1"
        >
          <CloseButton
            active={false}
            buttonChildren=""
            buttonStyles="button closeButton "
            clickAction={[Function]}
          />
          <span
            className="title "
          >
            TEST TOOLTIP
          </span>
        </div>
        <Arrow
          addArrow={true}
          arrowRotate={
            Object {
              "coordX": "center",
              "coordY": "top",
            }
          }
          arrowStyles=" tooltipArrow"
          arrowWrapStyles=" arrowWrap"
          key="2"
        >
          <div
            className=" arrowWrap"
          >
            <div
              className="arrow arrowcentertop  tooltipArrow"
            />
          </div>
        </Arrow>
      </div>
    </Transition>
  </CSSTransition>
</ToolTip>
`;

exports[`<Tooltip test /> should render Tooltip by manual coords coords fix props 1`] = `
<ToolTip
  actionOnEntered={[Function]}
  actionOnExit={[Function]}
  addArrow={true}
  animProps={
    Object {
      "className": "tooltip",
      "timeExit": 1000,
      "timeIn": 30,
    }
  }
  closeButton={
    Object {
      "actionOnClose": [Function],
      "active": false,
      "buttonChildren": "",
    }
  }
  customCoords={
    Object {
      "active": false,
      "left": "",
      "top": "",
    }
  }
  manualCoodsFix={
    Object {
      "fixX": -2,
      "fixY": -3,
    }
  }
  overrideStyles={
    Object {
      "arrowWrap": "",
      "button": "",
      "container": "",
      "title": "",
      "tooltipArrow": "",
      "wrap": "",
    }
  }
  position={
    Object {
      "x": "center",
      "y": "top",
    }
  }
>
  <CSSTransition
    classNames="tooltip"
    in={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onExited={[Function]}
    timeout={
      Object {
        "enter": 30,
        "exit": 1000,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 30,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <div
        className="tooltipWrap "
        id="#react-tooltip-tooltip-test"
        style={
          Object {
            "flexDirection": "column",
            "left": -3,
            "pointerEvents": "none",
            "top": -122,
            "transform": "translate(-50%, 0%)",
          }
        }
      >
        <div
          className=" tooltipContainer"
          key="1"
        >
          <CloseButton
            active={false}
            buttonChildren=""
            buttonStyles="button closeButton "
            clickAction={[Function]}
          />
          <span
            className="title "
          >
            TEST TOOLTIP
          </span>
        </div>
        <Arrow
          addArrow={true}
          arrowRotate={
            Object {
              "coordX": "center",
              "coordY": "top",
            }
          }
          arrowStyles=" tooltipArrow"
          arrowWrapStyles=" arrowWrap"
          key="2"
        >
          <div
            className=" arrowWrap"
          >
            <div
              className="arrow arrowcentertop  tooltipArrow"
            />
          </div>
        </Arrow>
      </div>
    </Transition>
  </CSSTransition>
</ToolTip>
`;

exports[`<Tooltip test /> should render Tooltip by position props 1`] = `
<ToolTip
  actionOnEntered={[Function]}
  actionOnExit={[Function]}
  addArrow={true}
  animProps={
    Object {
      "className": "tooltip",
      "timeExit": 1000,
      "timeIn": 30,
    }
  }
  closeButton={
    Object {
      "actionOnClose": [Function],
      "active": false,
      "buttonChildren": "",
    }
  }
  customCoords={
    Object {
      "active": false,
      "left": "",
      "top": "",
    }
  }
  overrideStyles={
    Object {
      "arrowWrap": "",
      "button": "",
      "container": "",
      "title": "",
      "tooltipArrow": "",
      "wrap": "",
    }
  }
  position={
    Object {
      "x": "center",
      "y": "top",
    }
  }
>
  <CSSTransition
    classNames="tooltip"
    in={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onExited={[Function]}
    timeout={
      Object {
        "enter": 30,
        "exit": 1000,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 30,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <div
        className="tooltipWrap "
        id="#react-tooltip-tooltip-test"
        style={
          Object {
            "flexDirection": "column",
            "left": 0,
            "pointerEvents": "none",
            "top": -120,
            "transform": "translate(-50%, 0%)",
          }
        }
      >
        <div
          className=" tooltipContainer"
          key="1"
        >
          <CloseButton
            active={false}
            buttonChildren=""
            buttonStyles="button closeButton "
            clickAction={[Function]}
          />
          <span
            className="title "
          >
            TEST TOOLTIP
          </span>
        </div>
        <Arrow
          addArrow={true}
          arrowRotate={
            Object {
              "coordX": "center",
              "coordY": "top",
            }
          }
          arrowStyles=" tooltipArrow"
          arrowWrapStyles=" arrowWrap"
          key="2"
        >
          <div
            className=" arrowWrap"
          >
            <div
              className="arrow arrowcentertop  tooltipArrow"
            />
          </div>
        </Arrow>
      </div>
    </Transition>
  </CSSTransition>
</ToolTip>
`;

exports[`<Tooltip test /> should render Tooltip with Close Button 1`] = `
<ToolTip
  actionOnEntered={[Function]}
  actionOnExit={[Function]}
  addArrow={true}
  animProps={
    Object {
      "className": "tooltip",
      "timeExit": 1000,
      "timeIn": 30,
    }
  }
  closeButton={
    Object {
      "active": true,
      "buttonChildren": "XXX",
    }
  }
  customCoords={
    Object {
      "active": false,
      "left": "",
      "top": "",
    }
  }
  overrideStyles={
    Object {
      "arrowWrap": "",
      "button": "",
      "container": "",
      "title": "",
      "tooltipArrow": "",
      "wrap": "",
    }
  }
  position={
    Object {
      "x": "center",
      "y": "top",
    }
  }
>
  <CSSTransition
    classNames="tooltip"
    in={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onExited={[Function]}
    timeout={
      Object {
        "enter": 30,
        "exit": 1000,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 30,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <div
        className="tooltipWrap "
        id="#react-tooltip-tooltip-test"
        style={
          Object {
            "flexDirection": "column",
            "left": 0,
            "pointerEvents": "all",
            "top": -120,
            "transform": "translate(-50%, 0%)",
          }
        }
      >
        <div
          className=" tooltipContainer"
          key="1"
        >
          <CloseButton
            active={true}
            buttonChildren="XXX"
            buttonStyles="button closeButton "
            clickAction={[Function]}
          >
            <button
              className="button closeButton "
              onClick={[Function]}
              type="button"
            >
              XXX
            </button>
          </CloseButton>
          <span
            className="title "
          >
            TEST TOOLTIP
          </span>
        </div>
        <Arrow
          addArrow={true}
          arrowRotate={
            Object {
              "coordX": "center",
              "coordY": "top",
            }
          }
          arrowStyles=" tooltipArrow"
          arrowWrapStyles=" arrowWrap"
          key="2"
        >
          <div
            className=" arrowWrap"
          >
            <div
              className="arrow arrowcentertop  tooltipArrow"
            />
          </div>
        </Arrow>
      </div>
    </Transition>
  </CSSTransition>
</ToolTip>
`;

exports[`<Tooltip test /> should render Tooltip with Close Button. By click on it should remove Tooltip at all 1`] = `
<ToolTip
  actionOnEntered={[Function]}
  actionOnExit={[Function]}
  addArrow={true}
  animProps={
    Object {
      "className": "tooltip",
      "timeExit": 1000,
      "timeIn": 30,
    }
  }
  closeButton={
    Object {
      "active": true,
      "buttonChildren": "XXX",
    }
  }
  customCoords={
    Object {
      "active": false,
      "left": "",
      "top": "",
    }
  }
  overrideStyles={
    Object {
      "arrowWrap": "",
      "button": "",
      "container": "",
      "title": "",
      "tooltipArrow": "",
      "wrap": "",
    }
  }
  position={
    Object {
      "x": "center",
      "y": "top",
    }
  }
>
  <CSSTransition
    classNames="tooltip"
    in={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onExited={[Function]}
    timeout={
      Object {
        "enter": 30,
        "exit": 1000,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 30,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <div
        className="tooltipWrap "
        id="#react-tooltip-tooltip-test"
        style={
          Object {
            "flexDirection": "column",
            "left": 0,
            "pointerEvents": "all",
            "top": -120,
            "transform": "translate(-50%, 0%)",
          }
        }
      >
        <div
          className=" tooltipContainer"
          key="1"
        >
          <CloseButton
            active={true}
            buttonChildren="XXX"
            buttonStyles="button closeButton "
            clickAction={[Function]}
          >
            <button
              className="button closeButton "
              onClick={[Function]}
              type="button"
            >
              XXX
            </button>
          </CloseButton>
          <span
            className="title "
          >
            TEST TOOLTIP
          </span>
        </div>
        <Arrow
          addArrow={true}
          arrowRotate={
            Object {
              "coordX": "center",
              "coordY": "top",
            }
          }
          arrowStyles=" tooltipArrow"
          arrowWrapStyles=" arrowWrap"
          key="2"
        >
          <div
            className=" arrowWrap"
          >
            <div
              className="arrow arrowcentertop  tooltipArrow"
            />
          </div>
        </Arrow>
      </div>
    </Transition>
  </CSSTransition>
</ToolTip>
`;

exports[`<Tooltip test /> should render Tooltip with Close Button. By click on it should remove Tooltip at all 2`] = `
<ToolTip
  actionOnEntered={[Function]}
  actionOnExit={[Function]}
  addArrow={true}
  animProps={
    Object {
      "className": "tooltip",
      "timeExit": 1000,
      "timeIn": 30,
    }
  }
  closeButton={
    Object {
      "active": true,
      "buttonChildren": "XXX",
    }
  }
  customCoords={
    Object {
      "active": false,
      "left": "",
      "top": "",
    }
  }
  overrideStyles={
    Object {
      "arrowWrap": "",
      "button": "",
      "container": "",
      "title": "",
      "tooltipArrow": "",
      "wrap": "",
    }
  }
  position={
    Object {
      "x": "center",
      "y": "top",
    }
  }
>
  <CSSTransition
    classNames="tooltip"
    in={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onExited={[Function]}
    timeout={
      Object {
        "enter": 30,
        "exit": 1000,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 30,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <div
        className="tooltipWrap "
        id="#react-tooltip-"
        style={
          Object {
            "flexDirection": "column",
            "left": 0,
            "pointerEvents": "all",
            "top": 0,
            "transform": "translate(-50%, 0%)",
          }
        }
      >
        <div
          className=" tooltipContainer"
          key="1"
        >
          <CloseButton
            active={true}
            buttonChildren="XXX"
            buttonStyles="button closeButton "
            clickAction={[Function]}
          >
            <button
              className="button closeButton "
              onClick={[Function]}
              type="button"
            >
              XXX
            </button>
          </CloseButton>
          <span
            className="title "
          />
        </div>
        <Arrow
          addArrow={true}
          arrowRotate={
            Object {
              "coordX": "center",
              "coordY": "top",
            }
          }
          arrowStyles=" tooltipArrow"
          arrowWrapStyles=" arrowWrap"
          key="2"
        >
          <div
            className=" arrowWrap"
          >
            <div
              className="arrow arrowcentertop  tooltipArrow"
            />
          </div>
        </Arrow>
      </div>
    </Transition>
  </CSSTransition>
</ToolTip>
`;

exports[`<Tooltip test /> should render Tooltip without Arrow 1`] = `
<ToolTip
  actionOnEntered={[Function]}
  actionOnExit={[Function]}
  addArrow={false}
  animProps={
    Object {
      "className": "tooltip",
      "timeExit": 1000,
      "timeIn": 30,
    }
  }
  closeButton={
    Object {
      "actionOnClose": [Function],
      "active": false,
      "buttonChildren": "",
    }
  }
  customCoords={
    Object {
      "active": false,
      "left": "",
      "top": "",
    }
  }
  overrideStyles={
    Object {
      "arrowWrap": "",
      "button": "",
      "container": "",
      "title": "",
      "tooltipArrow": "",
      "wrap": "",
    }
  }
  position={
    Object {
      "x": "center",
      "y": "top",
    }
  }
>
  <CSSTransition
    classNames="tooltip"
    in={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onExited={[Function]}
    timeout={
      Object {
        "enter": 30,
        "exit": 1000,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 30,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <div
        className="tooltipWrap "
        id="#react-tooltip-tooltip-test"
        style={
          Object {
            "flexDirection": "column",
            "left": 0,
            "pointerEvents": "none",
            "top": -120,
            "transform": "translate(-50%, 0%)",
          }
        }
      >
        <div
          className=" tooltipContainer"
          key="1"
        >
          <CloseButton
            active={false}
            buttonChildren=""
            buttonStyles="button closeButton "
            clickAction={[Function]}
          />
          <span
            className="title "
          >
            TEST TOOLTIP
          </span>
        </div>
        <Arrow
          addArrow={false}
          arrowRotate={
            Object {
              "coordX": "center",
              "coordY": "top",
            }
          }
          arrowStyles=" tooltipArrow"
          arrowWrapStyles=" arrowWrap"
          key="2"
        />
      </div>
    </Transition>
  </CSSTransition>
</ToolTip>
`;

exports[`<Tooltip test /> should render basic Tooltip with Title and Arrow included 1`] = `
<ToolTip
  actionOnEntered={[Function]}
  actionOnExit={[Function]}
  addArrow={true}
  animProps={
    Object {
      "className": "tooltip",
      "timeExit": 1000,
      "timeIn": 30,
    }
  }
  closeButton={
    Object {
      "actionOnClose": [Function],
      "active": false,
      "buttonChildren": "",
    }
  }
  customCoords={
    Object {
      "active": false,
      "left": "",
      "top": "",
    }
  }
  overrideStyles={
    Object {
      "arrowWrap": "",
      "button": "",
      "container": "",
      "title": "",
      "tooltipArrow": "",
      "wrap": "",
    }
  }
  position={
    Object {
      "x": "center",
      "y": "top",
    }
  }
>
  <CSSTransition
    classNames="tooltip"
    in={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onExited={[Function]}
    timeout={
      Object {
        "enter": 30,
        "exit": 1000,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 30,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <div
        className="tooltipWrap "
        id="#react-tooltip-tooltip-test"
        style={
          Object {
            "flexDirection": "column",
            "left": 0,
            "pointerEvents": "none",
            "top": -120,
            "transform": "translate(-50%, 0%)",
          }
        }
      >
        <div
          className=" tooltipContainer"
          key="1"
        >
          <CloseButton
            active={false}
            buttonChildren=""
            buttonStyles="button closeButton "
            clickAction={[Function]}
          />
          <span
            className="title "
          >
            TEST TOOLTIP
          </span>
        </div>
        <Arrow
          addArrow={true}
          arrowRotate={
            Object {
              "coordX": "center",
              "coordY": "top",
            }
          }
          arrowStyles=" tooltipArrow"
          arrowWrapStyles=" arrowWrap"
          key="2"
        >
          <div
            className=" arrowWrap"
          >
            <div
              className="arrow arrowcentertop  tooltipArrow"
            />
          </div>
        </Arrow>
      </div>
    </Transition>
  </CSSTransition>
</ToolTip>
`;
